### PR TITLE
Fix/field key from trails

### PR DESCRIPTION
**Pull request by acunniffe**
## Why
Adding extra information to the shape trails reduces the # of queries that have to run when interpreting diffs. 

## Validation
* [ ] CI passes
* [ ] Verified in staging
* etc...

### CODE REVIEW

File: workspaces/diff-engine/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "postinstall": "node scripts/install",
     "preuninstall": "node scripts/uninstall",
     "test": "cd tests && tap ./lib",
-    "ws:build-binaries": "scripts/build-binaries.sh",
+    "ws:build": "cargo build",
```
Comment (devdoshi) [inline workspaces/diff-engine/package.json:8]: @JaapRood is this right?

File: workspaces/ui-v2/src/lib/shape-trail-parser.ts [hunk 2]
```diff
@@ -73,10 +66,9 @@ export async function shapeTrailParserLastId(
       return {
         lastObject: lastObjectOption
           ? lastObjectOption[0]
-          : field.parentObjectId,
+          : fieldTrail.parentObjectShapeId,
```
Comment (devdoshi) [inline workspaces/ui-v2/src/lib/shape-trail-parser.ts:69]: Could this just always use the trail now?

File: workspaces/diff-engine/package.json (hunk 1 above)
Comment (JaapRood) [reply workspaces/diff-engine/package.json:8]: Yeah that should be fine.

File: workspaces/ui-v2/src/lib/shape-trail-parser.ts (hunk 2 above)
Comment (acunniffe) [reply workspaces/ui-v2/src/lib/shape-trail-parser.ts:69]: this is needed for new / nested fields.

**Pull request merged**